### PR TITLE
Add org.kde.kommit

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://invent.kde.org/teams/flathub/issues/-/issues/14
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by Mailaender** [comment]
Can't build right now, as https://release-monitoring.org/ is down.

**Comment by hfiguiere** [comment]
cc @flathub/kde

File: org.kde.kommit.json [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                        "type": "anitya",
+                        "project-id": 4941,
+                        "stable-only": true,
+                        "url-template": "https://prdownloads.sourceforge.net/tcl/tcl$version-src.tar.gz"
+                    }
+                }
+            ]
+        },
+        {
+            "name": "tk",
```
Comment (hfiguiere) [inline org.kde.kommit.json:54]: is a tk app intended to be run from inside this flatpak?

File: org.kde.kommit.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        "--share=ipc",
+        "--share=network",
+        "--device=dri",
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--system-talk-name=org.freedesktop.UDisks2"
+    ],
+    "modules": [
+        {
+            "name": "kommit",
```
Comment (hfiguiere) [inline org.kde.kommit.json:18]: this should be last in the modules list.

File: org.kde.kommit.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                    "url": "https://github.com/git-lfs/git-lfs/releases/download/v3.4.0/git-lfs-linux-amd64-v3.4.0.tar.gz",
+                    "sha256": "60b7e9b9b4bca04405af58a2cd5dff3e68a5607c5bc39ee88a5256dd7a07f58c",
+                    "x-checker-data": {
+                        "type": "json",
+                        "url": "https://api.github.com/repos/git-lfs/git-lfs/releases/latest",
+                        "url-query": ".assets[] | select(.name==\"git-lfs-linux-amd64-\" + $version + \".tar.gz\") | .browser_download_url",
+                        "version-query": ".tag_name"
+                    },
+                    "only-arches": [
+                        "x86_64"
```
Comment (hfiguiere) [inline org.kde.kommit.json:73]: why not on aarch64?

File: org.kde.kommit.json (hunk 2 above)
Comment (Mailaender) [reply org.kde.kommit.json:18]: No, git is not a compile time dependency.

**Comment by hfiguiere** [comment]
bot, build org.kde.kommit

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/org.kde.kommit.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• tag-missing           : <release> required
Validation of files failed
```

**Comment by Mailaender** [comment]
bot, build org.kde.kommit

File: org.kde.kommit.json [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                    ]
+                },
+                {
+                    "type": "archive",
+                    "url": "https://github.com/git-lfs/git-lfs/releases/download/v3.4.0/git-lfs-linux-arm64-v3.4.0.tar.gz",
+                    "sha256": "e97c477981a9b6a40026cadc1bf005541d973fc32df2de2f398643b15df6b5c6",
```
Comment (hfiguiere) [inline org.kde.kommit.json:83]: apparently this checksum is incorrect

File: org.kde.kommit.json (hunk 2 above)
Comment (hfiguiere) [reply org.kde.kommit.json:18]: doesn't matter. it's the convention.

**Comment by Mailaender** [comment]
bot, build org.kde.kommit

**Comment by tsdgeos** [comment]
This is not very useful since it doesn't seem to be able to access my ssh keys

So any operation i try fails with 

git@gitlab.freedesktop.org: Permission denied (publickey).

**Comment by Justinzobel** [comment]
Yeah I think it's because it can't access the ssh folder which I think is a good/bad thing. I saw the ssh error in konsole when running it and it stopped until I forced it to close.

**Comment by hfiguiere** [comment]
Shouldn't this work with the secrets service DBus API?

**Comment by Mailaender** [comment]
Added the SSH auth socket.

**Comment by Mailaender** [comment]
bot, build org.kde.kommit

File: org.kde.kommit.json (hunk 2 above)
Comment (travier) [reply org.kde.kommit.json:19]: We still need this

**Comment by travier** [comment]
We probably need to give it GPG access too as some people sign their commits that way.

**Comment by Mailaender** [comment]
Added that as well.

**Comment by Mailaender** [comment]
bot, build org.kde.kommit

**Comment by travier** [comment]
https://github.com/flathub/flathub/pull/4401/files#r1291086950

**Comment by Mailaender** [comment]
bot, build org.kde.kommit

**Comment by tsdgeos** [comment]
> Shouldn't this work with the secrets service DBus API?

No such thing on my system.

**Comment by tsdgeos** [comment]
I woud really prefer if we put this on beta.

For example when trying to clone from an unkonwn host, it will ask

The authenticity of host 'gitlab.freedesktop.org (147.75.198.156)' can't be established.
ED25519 key fingerprint is SHA256:i4LwjlmsgSrc3aWq3aVaDa8VQs6Fqq4nRBuFdZ6A1tE.
This key is not known by any other names.
Are you sure you want to continue connecting (yes/no/[fingerprint])?

on the command line, that-s a bug that exists in kommit upstream too, but this can be solved in kommit upstream by doing things from the git command line first and it will learn about the host, but that's something that you can't do in flatpak i think? 

Or is there a way to share the known_hosts file from the host to the flatpak app?

**Comment by travier** [comment]
We could explicitly share the known host file in the Flatpak (as RW?).

**Comment by Mailaender** [comment]
bot, build org.kde.kommit

**Comment by hfiguiere** [comment]
did you test it? AFAIK `--filesystem` should be a *directory*

**Comment by Mailaender** [comment]
No, sorry I switched computers from Arch to an Ubuntu LTS derivative where I get
> error: Unknown socket type gpg-agent, valid types are: x11, wayland, pulseaudio, session-bus, system-bus, fallback-x11, ssh-auth, pcsc, cups

when I try to build.

**Comment by Mailaender** [comment]
bot, build org.kde.kommit

**Comment by tsdgeos** [comment]
There's still one more small wrinkle (sorry, testing as we go)

If i do file->clone

and then on folder manually type /home/tsdgeos/lololo  (where /home/tsdgeos is my home) and then press OK, it succeeds, BUT the clone is nowhere to be found ^_^

If i use the file selector button then i get a path like /run/user/1000/doc/416f60f/tsdgeos/lololo and then when pressing OK it works.

So it's not totally broken but "half" broken in the sense that:
 * manually typing the path doesn't work
 * selecting the path in the folder gives you a "weird" path that the user may be scared of

**Comment by hfiguiere** [comment]
maybe the only way is `--filesystem=home`...

**Comment by Mailaender** [comment]
We did the same for https://github.com/flathub/com.github.git_cola.git-cola/blob/master/com.github.git_cola.git-cola.yml#L17

File: org.kde.kommit.json [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            "name": "git-lfs",
+            "buildsystem": "simple",
+            "build-commands": [
+                "PREFIX=${FLATPAK_DEST} ./install.sh"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "strip-components": 1,
+                    "url": "https://github.com/git-lfs/git-lfs/releases/download/v3.4.0/git-lfs-linux-amd64-v3.4.0.tar.gz",
```
Comment (tsdgeos) [inline org.kde.kommit.json:56]: Why are we downloading a binary instead of compiling it?

Comment (Mailaender) [reply org.kde.kommit.json:56]: Ask the authors of https://github.com/flathub/org.flatpak.Builder/blob/master/org.flatpak.Builder.json#L238 I assume packaging Go is not well supported yet.

Comment (tsdgeos) [reply org.kde.kommit.json:56]: I don't know why I would ask people unrelated to this MR, they are not proposing anything.

https://github.com/fyne-io/flatpak_demo/blob/main/io.fyne.flatpak_demo.yml seems to say building a go app just works.

Comment (Mailaender) [reply org.kde.kommit.json:56]: I mean, they probably had their reasons to choose that path in an official @flathub package.

Comment (tsdgeos) [reply org.kde.kommit.json:56]: What do you mean "an official flathub package"?

Comment (hfiguiere) [reply org.kde.kommit.json:56]: My understanding is that git-lfs had botched vendoring the Go packages, so build kept failing in the pursuit of always having the latest package. (This is not an approval)

`org.flatpak.Builder` is maintained by the Flathub maintainers as it is the one used on the builders.

Comment (tsdgeos) [reply org.kde.kommit.json:56]: > org.flatpak.Builder is maintained by the Flathub maintainers as it is the one used on the builders.

This is starting to discussing the smell of clours, but that doesn't really make it a more official package than say Okular.

> My understanding is that git-lfs had botched vendoring the Go packages, so build kept failing in the pursuit of always having the latest package. (This is not an approval)

I would really like if there was an attempt at making this work and someone could show me the error before we accept defeat and include a "random binary" in this package.

**Comment by Mailaender** [comment]
bot, build org.kde.kommit

**Comment by tsdgeos** [comment]
So you are not even try to get it to build?

Ok for me.

**Comment by Mailaender** [comment]
I understand your objection of avoiding binary blobs even though it is allowed on Flathub. I am not skilled enough to build it from source. My experience with Go is very limited.

**Comment by hfiguiere** [comment]
ok to merge, stable or beta?

git-lfs can still be fixed afterwards

@Mailaender it's a good way to learn.

**Comment by travier** [comment]
Let's go for stable and we'll create an issue for Git LFS support

**Pull request closed without merging**